base/abstractarray.jl

```diff
@@ -1737,6 +1737,7 @@ function mapslices(f, A::AbstractArray, dims::AbstractVector)
 
     Aslice = A[idx...]
     r1 = f(Aslice)
+    safe_for_reuse = isa(r1, Number) || (isa(r1, AbstractArray) && eltype(r1) <: Number)
```

Comment:
Yes, it's "safe" for any object that couldn't possibly contain the passed array.